CMakeLists.txt

```diff
@@ -162,13 +163,15 @@ install(TARGETS cable RUNTIME)
 if(CABLE_MPI)
     add_executable(
         cable-mpi
+        src/offline/cable_driver_init.F90
```

Comment:
I don't think this works as intended. Basically there are two rules to generate the same object file and associated module file, one for serial and one for MPI. This means that when they are recompiled the second time, they object file from the first compilation is overwritten.

Compilation when using Makefiles works, but more by accident than by design. On the other hand, Ninja simply refuses to compile the files and rightly so.

I think the correct solution here is to have all the serial and MPI specific objects and module files being generated in different folders.

Reply:
For reference [this issue](https://gitlab.kitware.com/cmake/cmake/-/issues/22898) looks like it describes what is happening in our case. It provides a workaround using the [Fortran_MODULE_DIRECTORY](https://cmake.org/cmake/help/latest/prop_tgt/Fortran_MODULE_DIRECTORY.html) target property. Micael has proposed an alternative way to structure the CMakeLists.txt so we avoid this issue entirely.

Reply:
I would suggest to merge this PR as is and I'll then create a new PR with further changes that will have the side-effect of fixing this issues.

Reply:
New branch with proposed changes here: https://github.com/CABLE-LSM/CABLE/tree/introduce_mpi_group Relevant commit: b9e6056e636fe077ad2ddabd53a5f3f1d3661a2d

This still needs to some to be properly documented.